pandas-stubs/_typing.pyi

```diff
@@ -404,6 +404,29 @@ Function: TypeAlias = np.ufunc | Callable[..., Any]
 # shared HashableT and HashableT#. This one can be used if the identical
 # type is need in a function that uses GroupByObjectNonScalar
 _HashableTa = TypeVar("_HashableTa", bound=Hashable)
+ByT = TypeVar(
+    "ByT",
+    str,
+    bytes,
+    datetime.date,
+    datetime.datetime,
+    datetime.timedelta,
+    np.datetime64,
+    np.timedelta64,
+    bool,
+    int,
+    float,
+    complex,
+    Timestamp,
+    Timedelta,
```

Comment:
You can delete all of these as they are in `Scalar`

Reply:
That's what I had at the start, but then mypy will complain about the `Scalar` overload on `DataFrame.groupby`. There is a subtle difference between `Union[DataframeGroupBy[str], ...]` and `DataFrameGroupBy[Union[str, ...]]` when matching overloads. This seemed like the most ergonomic way to still allow that to happen.

Reply:
Yes, I see what you're saying.  Apparently, if you have a parameter of `T` on a method that returns `X[T]` where `X` is `Generic` and `T` is a `TypeVar`, `T` doesn't get bound correctly.

Oh, well.